manifests/init.pp

```diff
@@ -73,6 +75,7 @@
   Boolean $generate = true,
   Boolean $regenerate = false,
   Boolean $deploy = true,
+  String[10] $ca_key_password = $certs::params::ca_key_password,
```

Comment:
Can you keep it on the line where it used to be? So between `ca_cert_stripped` and `ca_key_password_file`?

Reply:
This is new because `ca_key_password` was not in the parameters for the class previously.

Reply:
I'm debating having a default or not. Having it means it ends up in our installer's answer files. Not sure if that's a benefit or a downside. @ehelms?

I also wonder what happens if you change the password parameter. I fear it would break since there's no support to change the password.

Reply:
I'm a bit confused. I do not like having it as a default and showing up in the answers file. This currently already has a default coming from params.pp?

Reply:
The diff in GH isn't clear (I also failed at that at first). Right now it's in the body of the class. This PR moves it to a class parameter (with indirection via params since Kafo can't deal with calling functions as defaults).

Reply:
@ehelms can we revisit this? I still think this is a good direction.

I'm still leaning to `undef` and also support `Sensitive`. So `Variant[Undef, Sensitive[String[10]], String[10]]] $ca_key_password = undef` (or `Optional[Variant[Sensitive[String[10]], String[10]]]` which is equivalent).

Reply:
If I understand correctly, for an existing setup, if a new ca_key_password is provided this will force a new key to be generated, which forces a new CA to be generated which cascades and forces all smart-proxy / Capsules and clients to need to be updated. 

Will the code properly handle that? I think we need a test for it. How do we help users not break their installations accidentally?

Reply:
One data point that just came up today: if puppet-agent is removed then the cache is wiped. If the ssl-build directory isn't wiped, then there's a mismatch. If we have the value as an explicit parameter, the value is stored in the answer file and there wouldn't be an issue. So an argument against `undef` as default and in favor of the current code.